source/includes/write/transactions.cpp

```diff
@@ -0,0 +1,141 @@
+                    std::cout << "Transaction aborted. Caught exception during transaction." << std::endl;
+                    // If transient error, retry the whole transaction.
+                    if (oe.has_error_label("TransientTransactionError")) {
+                        std::cout << "TransientTransactionError, retrying transaction ..." << std::endl;
+                        continue;
+                    } else {
+                        throw oe;
+                    }
+                }
+            }
+        };
+
+        // Incorporate UnknownTransactionCommitResult retry logic
+        auto commit_with_retry = [](mongocxx::client_session& session) {
+            while (true) {
+                try {
+                    session.commit_transaction();  // Uses write concern set at transaction start.
+                    std::cout << "Transaction committed." << std::endl;
+                    break;
+                } catch (const mongocxx::operation_exception& oe) {
+                    // Can retry commit
+                    if (oe.has_error_label("UnknownTransactionCommitResult")) {
+                        std::cout << "UnknownTransactionCommitResult, retrying commit operation ..." << std::endl;
+                        continue;
+                    } else {
+                        std::cout << "Error during commit ..." << std::endl;
+                        throw oe;
+                    }
+                }
+            }
+        };
+
+        auto run_transaction = [&](mongocxx::client_session& session) {
+            auto& client = session.client(); 
+
+            // Define database and collection variables
+            auto db = client["sample_mflix"];
+            auto movies_collection = db["movies"];
+            auto comments_collection = db["comments"];
+
+            // Define an options instance to set the majority write concern for the transaction operations explicitly 
+            mongocxx::options::transaction opts;
+            mongocxx::write_concern wc;
+            wc.acknowledge_level(mongocxx::write_concern::level::k_majority);
+            opts.write_concern(wc);
+
+            session.start_transaction(opts);
+
+            // Attempt to insert documents into database collections
+            try {
+                movies_collection.insert_one(session, make_document(kvp("title", "Parasite")));
+                comments_collection.insert_one(session, make_document(kvp("name", "Rhaenyra Targaryen"), kvp("text", "Dracarys")));
+            } catch (const mongocxx::operation_exception& oe) {
+                std::cout << "Caught exception during transaction, aborting." << std::endl;
+                session.abort_transaction();
+                throw oe;
+            }
+
+            commit_with_retry(session);
+        };
+
+        // Start a client session
+        auto session = client.start_session();
+        
+        try {
+            run_transaction_with_retry(run_transaction, session);
```

Comment:
Q: is this function supposed to be `run_with_retry`?

Reply:
Yes! Thanks :)